helper/helper_test.go

```diff
@@ -98,4 +265,235 @@ var _ = Describe("helper tests", func() {
+				{100, 0},
+				{200, 0},
+				{200, 1},
+				{200, 2},
+			}
+
+			expectedScenario := []helperTestStep{
+				{},                               // placeholder as steps start from 1
+				{"mock_100_0", true, -1, false},  // Can open pipe for table 100, check_skip_file shall not be called
+				{"mock_200_0", true, -1, false},  // Can open pipe for table 200, check_skip_file shall not be called
+				{"mock_200_1", false, 200, true}, // Can not open pipe for table 200, check_skip_file shall called, skip file exists
+				{"mock_200_2", true, -1, false},  // Went to the next batch, Can open pipe for table 200, check_skip_file shall not be called
+			}
+
+			helper := newHelperTest(oidBatch, expectedScenario)
+			err := doRestoreAgentInternal(helper, helper)
+			Expect(err).To(BeNil())
+
+		})
+		It("skips batches if skip file is discovered with single datafile", func() {
+			*singleDataFile = true
+			*isResizeRestore = false
+			*tocFile = testTocFile
+
+			// Although pure concept would be to mock TOC file as well, to keep things simpler
+			// let's use real TOC file here
+			writeTestTOC(testTocFile)
+			defer func() {
+				_ = os.Remove(*tocFile)
+			}()
+
+			oidBatch := []oidWithBatch{
+				{100, 0},
+				{200, 0},
+				{200, 1},
+				{200, 2},
+			}
+
+			expectedScenario := []helperTestStep{
+				{},                               // placeholder as steps start from 1
+				{"mock_100_0", true, -1, false},  // Can open pipe for table 100, check_skip_file shall not be called
+				{"mock_200_0", true, -1, false},  // Can open pipe for table 200, check_skip_file shall not be called
+				{"mock_200_1", false, 200, true}, // Can not open pipe for table 200, check_skip_file shall called, skip file exists
+				{"mock_200_2", true, -1, false},  // Went to the next batch, Can open pipe for table 200, check_skip_file shall not be called
+			}
+
+			helper := newHelperTest(oidBatch, expectedScenario)
+			err := doRestoreAgentInternal(helper, helper)
+			Expect(err).To(BeNil())
+		})
+		It("calls Wait in waitForPlugin doRestoreAgent for single data file", func() {
+			*singleDataFile = true
+			*isResizeRestore = false
+			*tocFile = testTocFile
+
+			// Although pure concept would be to mock TOC file as well, to keep things simpler
+			// let's use real TOC file here
+			writeTestTOC(testTocFile)
+			defer func() {
+				_ = os.Remove(*tocFile)
+			}()
+
+			oidBatch := []oidWithBatch{{100, 0}}
+			expectedScenario := []helperTestStep{{}, {"mock_100_0", true, -1, false}} // Some pipe shall be created, out of interest for this test although
+			helper := newHelperTest(oidBatch, expectedScenario)
+
+			err := doRestoreAgentInternal(helper, helper)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Check that plugin command's Wait was acually called and only once
+			Expect(helper.restoreData.waitCount).To(Equal(1))
+		})
+		It("calls waitForPlugin doRestoreAgent for resize and no single data file ", func() {
+			*singleDataFile = false
```

Comment:
Is this really done before every test?

Reply:
Sorry, could not understand the question. Perhaps yes, it shall be done, why you doubt ?

Reply:
Why do we need this line here if it is already done before every test?

Reply:
Here I am trying to express importance of this line for the particular test. If bold was available in go syntax, i'd use it.

Reply:
You may use `Expect` for it.

Reply:
Pardon? Expect?

Reply:
```go
Expect(*singleDataFile).To(Equal(false))
```